### PR TITLE
feat(module:pagination): support Global Config

### LINKED ISSUE #6042: Pagination nzPageSizeOptions Global Configuration
## What problem does this feature solve?
I have to  write nzPageSizeOptions in every Pagination Component if it has difference with default

## What does the proposed API look like?
```js
pagination: { nzPageSizeOptions: [10, 20, 30, 40] }
```

### PR DESCRIPTION
**Pull request by Xie-Jay**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: resolve #6042 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/pagination/doc/index.zh-CN.md [hunk 1]
```diff
@@ -43,3 +43,12 @@ import { NzPaginationModule } from 'ng-zorro-antd/pagination';
 | `[nzHideOnSinglePage]` | 只有一页时是否隐藏分页器 | `boolean` | `false` |
 | `(nzPageIndexChange)` | 页码改变的回调 | `EventEmitter<number>` | - |
 | `(nzPageSizeChange)` | 每页条数改变的回调 | `EventEmitter<number>` | - |
+
+### 全局配置
```
Comment (vthinkxie) [inline components/pagination/doc/index.zh-CN.md:47]: 建议和其他组件的文档格式保持一致，在表格再增加一列是否支持全局

File: components/pagination/pagination.component.ts [hunk 2]
```diff
@@ -102,10 +104,24 @@ export class NzPaginationComponent implements OnInit, OnDestroy, OnChanges {
   showPagination = true;
   locale!: NzPaginationI18nInterface;
   size: 'default' | 'small' = 'default';
+  pageSizeOptions: number[] = [10, 20, 30, 40];
 
   private destroy$ = new Subject<void>();
   private total$ = new ReplaySubject<number>(1);
 
+  private onConfigChange(): void {
```
Comment (vthinkxie) [inline components/pagination/pagination.component.ts:112]: 建议参照其他组件的 @WithConfig 用法，可以大幅度简化代码实现

**Review by vthinkxie — CHANGES_REQUESTED** [review]
commit规范请遵循 angular commit format
例如 `feat(module:pagination): support nzPageSizeOptions global configuration`

代码实现部分也有一些建议，非常感谢提交代码

File: components/pagination/doc/index.zh-CN.md (hunk 1 above)
Comment (Xie-Jay) [reply components/pagination/doc/index.zh-CN.md]: 我没早到表格有支持全局的例子

File: components/pagination/pagination.component.ts (hunk 2 above)
Comment (Xie-Jay) [reply components/pagination/pagination.component.ts]: 已做修改[1c160a69f947b9c033f08d1e958088288b169131](https://github.com/NG-ZORRO/ng-zorro-antd/pull/6043/commits/1c160a69f947b9c033f08d1e958088288b169131)

**Review by hsuanxyz — APPROVED** [review]

**Pull request merged**